app/features/notifications/notification-hooks.ts

```diff
@@ -0,0 +1,185 @@
+import type { RealtimePostgresChangesPayload } from '@supabase/supabase-js';
+import {
+  type UseSuspenseQueryResult,
+  useMutation,
+  useQueryClient,
+  useSuspenseQuery,
+} from '@tanstack/react-query';
+import { useSupabaseRealtimeSubscription } from '~/lib/supabase/supabase-realtime';
+import { useLatest } from '~/lib/use-latest';
+import { type AgicashDbNotification, agicashDb } from '../agicash-db/database';
+import { useUserRef } from '../user/user-hooks';
+import type { Notification, NotificationType } from './notification';
+import { useNotificationRepository } from './notification-repository';
+
+const notificationsQueryKey = 'notifications';
+
+/**
+ * Returns a suspense query that returns all notifications of the given type.
+ */
+export function useNotifications<
+  T extends NotificationType = NotificationType,
+>(select: { type: T }): UseSuspenseQueryResult<Notification<T>[]> {
+  const notificationRepository = useNotificationRepository();
+  const userRef = useUserRef();
+
+  return useSuspenseQuery({
+    queryKey: [notificationsQueryKey, select.type],
+    queryFn: () =>
+      notificationRepository.list({
+        userId: userRef.current.id,
+        type: select.type,
+      }),
+    staleTime: 0,
+    refetchOnMount: 'always',
+    select: (data) =>
+      data.filter((n): n is Notification<T> => n.type === select.type),
+  });
+}
+
+export function useNotificationByTransactionId(transactionId: string) {
+  const notificationRepository = useNotificationRepository();
+
+  return useSuspenseQuery({
+    queryKey: [notificationsQueryKey, 'byTransactionId', transactionId],
+    queryFn: () => notificationRepository.getByTransactionId(transactionId),
+  });
+}
+
+export function useHasNotifications(select: {
+  type: NotificationType;
```

Comment:
also why filter by type? assuming we have more notification types later, wouldn't you want to show unread notification mark in ui if there is at least one unread notification of any type?